neos/neos-development-collection
# BUGFIX: Tags and collections of assets are lost when exporting to Sites.xml

see also #2503




**What I did**

Included tags and collections of assets in exports to `Sites.xml` and made sure they are imported properly.

**How I did it**

* Added converters from tags and asset collections to arrays:
  * For tags, the child property `assetCollections` is ignored because this would lead to an infinite loop.
  * For collections, the child property `assets` is ignored because this would lead to an infinite loop.
* Added a converter from array to tag. See the comment in https://github.com/neos/neos-development-collection/commit/3c5a4bb0d09bea16bcfdeceea6dc397adae662a9#diff-5d70864d51a35d6ab9fbb90ae189da11R57 for the reason why this is needed.
* For asset collections the generic converter from arrays suffices.

**How to verify it**

1. Add some tags and collections and assign some assets to them.
2. Export to `Sites.xml`.
3. Do a clean import.
4. Check that the tags and collections are preserved.

**Notes**
1. Of course, tags and collections not accessible from content are lost because the assets are lost.
2. I tried to test as many edge cases as possible but I'm not sure I have caught all of them. However, a clean import should work in any case so I don't expect any data loss even if some edge cases are not handled properly.
